viamonvif/device/device.go

```diff
@@ -286,6 +301,38 @@ func (dev *Device) GetStreamURI(ctx context.Context, profile onvif.Profile, cred
 	return uri, nil
 }
 
+// GetSnapshotURI is a request to the GetSnapshotURI onvif endpoint.
+func (dev *Device) GetSnapshotURI(ctx context.Context, token onvif.ReferenceToken, creds Credentials) (*url.URL, error) {
+	dev.logger.Debugf("GetSnapshotUri token: %s", token)
+	body, err := dev.callMedia(ctx, GetSnapshotURI{
+		ProfileToken: token,
+	})
+	if err != nil {
+		return nil, err
+	}
+	dev.logger.Debugf("GetSnapshotUri response: %v", string(body))
+	var snapshotURI getSnapshotURIResponse
+	if err := xml.NewDecoder(bytes.NewReader(body)).Decode(&snapshotURI); err != nil {
+		return nil, fmt.Errorf("failed to get snapshot URL for token %s: %w", token, err)
```

Comment:
is the token a secure value?

Reply:
This is a "reference token" (000, 001, ...) that refer to profiles that have already been setup from `GetProfiles`. 

However, not sure if it is that helpful to put the ref token in the log here -- can remove.

Reply:
We do have debug logs for profile name and corresponding reference token in GetProfiles so may still be useful to log token here.

Reply:
if it's not a secure value it can stay